File: app/src/test/java/net/rafaeltoledo/coverage/MainActivityUnitTest.java
package net.rafaeltoledo.coverage;

import androidx.test.core.app.ActivityScenario;
import androidx.test.ext.junit.runners.AndroidJUnit4;

import org.junit.Test;
import org.junit.runner.RunWith;

import static androidx.test.espresso.Espresso.onView;
import static androidx.test.espresso.action.ViewActions.click;
import static androidx.test.espresso.assertion.ViewAssertions.matches;
import static androidx.test.espresso.matcher.ViewMatchers.isDisplayed;
import static androidx.test.espresso.matcher.ViewMatchers.withId;
import static org.hamcrest.Matchers.not;

@RunWith(AndroidJUnit4.class)
public class MainActivityUnitTest {

    @Test
    public void shouldHideButtonAfterClick() {
        ActivityScenario.launch(MainActivity.class);

        onView(withId(R.id.hide)).perform(click());

        onView(withId(R.id.hide)).check(matches(not(isDisplayed())));
    }
}

File: app/src/test/kotlin/net/rafaeltoledo/coverage/MainActivityUnitTest.kt
package net.rafaeltoledo.coverage

import android.view.View
import android.widget.Button
import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.`is`
import org.junit.Test
import org.junit.runner.RunWith
import org.robolectric.Robolectric
import org.robolectric.RobolectricTestRunner

@RunWith(RobolectricTestRunner::class)
class MainActivityUnitTest {

    @Test
    fun shouldHideButtonAfterClick() {
        val activity = Robolectric.setupActivity(MainActivity::class.java)

        val button = activity.findViewById<Button>(R.id.hide)
        button.performClick()

        assertThat(button.visibility, `is`(View.GONE))
    }
}